contracts/contracts/vault/OETHBaseVaultCore.sol

```diff
@@ -59,4 +59,22 @@ contract OETHBaseVaultCore is OETHVaultCore {
         // Burn OTokens
         oUSD.burn(msg.sender, amount);
     }
+
+    // @inheritdoc OETHVaultCore
+    function _redeem(uint256 _amount, uint256 _minimumUnitAmount)
```

Comment:
I like how you've added the check in the internal function rather than 2 public facing `redeem` & `redeemAll` functions. 👍